### PR TITLE
Fix negative binomial's scaling

### PR DESCRIPTION
**Pull request by kashif**
Fixes issue #718

Also might explain issue #636

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by canerturkmen** [comment]
Hi @kashif, thank you so much for surfacing this. I can confirm that indeed, it seems like a bug. However I couldn't validate your fix as well. Do you have a proof / argument / test anywhere that I can check?

As a counterargument here, this is the "mean-dispersion" parameterization of the negative binomial. The previous version "overdisperses" the distribution while the fix does the inverse. I don't think either are correct.

**Comment by kashif** [comment]
thanks @canerturkmen So let see... in the current parametrisation, of `mu` and `alpha` the NB's mean is `mu` and variance is: `mu + alpha * mu**2`. 

So if we now want to sample from a NB scaled by `nu`, the mean has to become `mu = mu*nu` and the variance has to scale by `nu**2` in which case the transformation to alpha has to transform by:

```alpha =  alpha - (nu -1)/(nu*mu)```

The transformation I have here is not really correct either, but it mirrors the transformation of the paper and doesn't have the extra `+1`. I am happy to change it to the above version. What do you say?

**Comment by lostella** [comment]
> I am happy to change it to the above version. What do you say?

I agree that's how it should be. This also agrees with @canerturkmen 's intuition. I'd say we give it try it and check whether it solves the related issues? Maybe we they can be reproduced

**Comment by kashif** [comment]
ok cool! In our experiments this proposed transformation gave better neg. log likelihoods... I'll update this PR

**Comment by lostella** [comment]
@kashif maybe it's `alpha' = alpha + (nu-1)/(nu*mu)` with a plus?

**Comment by kashif** [comment]
@lostella yes just typo writing from my notes...

**Comment by canerturkmen** [comment]
Yup. That should be the one.

@kashif, out of interest, could you share which proposal gave better results and which paper does your initial reply refer to?

File: src/gluonts/distribution/neg_binomial.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def distribution(
         else:
             F = getF(mu)
             mu = F.broadcast_mul(mu, scale)
-            alpha = F.broadcast_div(alpha, F.sqrt(scale))
+            alpha = F.broadcast_add(alpha, F.broadcast_div(scale - 1, mu))
```
Comment (canerturkmen) [inline src/gluonts/distribution/neg_binomial.py:133]: Is this  `F.broadcast_div(scale - 1, scale*mu)` ?

**Comment by kashif** [comment]
@canerturkmen  This current change gave the best neg. log likelihood and the DeepAR paper section 4.3 ( https://www.sciencedirect.com/science/article/pii/S0169207019301888 ) has the initial change...

**Comment by kashif** [comment]
@lostella not sure about mxnet broadcast functions... perhaps I did something wrong, kindly change it if so, thanks!

**Comment by canerturkmen** [comment]
> @lostella not sure about mxnet broadcast functions... perhaps I did something wrong, kindly change it if so, thanks!

Got it now, thanks also for the pointers.

**Comment by kashif** [comment]
from my side once this issue is done, it would be a good time for a release a point release...

**Comment by kashif** [comment]
@canerturkmen would you be able to run some tests on your side with this scaling?

**Comment by canerturkmen** [comment]
> @canerturkmen would you be able to run some tests on your side with this scaling?

Yes, we'll give this another look @kashif. Thank you for your contribution.

**Pull request merged**